app/models/katello/system.rb

```diff
@@ -69,16 +62,18 @@ class System < Katello::Model
     before_update :update_foreman_host, :if => proc { |r| r.environment_id_changed? || r.content_view_id_changed? }
 
     scope :in_environment, ->(env) { where('environment_id = ?', env) unless env.nil? }
-    scope :completer_scope, ->(options) { readable(options[:organization_id]) }
     scope :by_uuids, ->(uuids) { where(:uuid => uuids) }
 
     scoped_search :on => :name, :complete_value => true
-    scoped_search :in => :environment, :on => :organization_id, :complete_value => true, :rename => :organization_id
+    scoped_search :in => :activation_keys, :on => :name, :complete_value => true, :rename => :activation_key
+    scoped_search :in => :content_view, :on => :name, :complete_value => true, :rename => :content_view
+    #this can't be added until http://projects.theforeman.org/issues/12239 is resolved
+    #scoped_search :in => :environment, :on => :name, :complete_value => true, :rename => :environment
 
     has_many :fact_values, :through => :foreman_host
     has_many :fact_names, :through => :fact_values
 
-    scoped_search :in => :fact_values, :on => :value, :in_key=> :fact_names, :on_key=> :name, :rename => :facts, :complete_value => true,
+    scoped_search :in => :fact_values, :on => :value, :in_key => :fact_names, :on_key => :name, :rename => :facts, :complete_value => true,
```

Comment:
Do you mind grouping all the scoped_search calls? They were scattered after the fact commit.